packages/fuels-programs/src/call_utils.rs

```diff
@@ -920,4 +920,84 @@ mod test {
             expected_asset_id_amounts
         )
     }
+
+    mod compute_calls_instructions_len {
+        use fuel_asm::Instruction;
+        use fuels_core::types::{enum_variants::EnumVariants, param_types::ParamType};
+
+        use crate::{call_utils::compute_calls_instructions_len, contract::ContractCall};
+
+        const BASE_INSTRUCTION_COUNT: usize = 5;
+
+        #[test]
+        fn test_simple() {
+            let call = ContractCall::new_with_random_id();
+            let instructions_len = compute_calls_instructions_len(&[call]).unwrap();
+            assert_eq!(instructions_len, Instruction::SIZE * BASE_INSTRUCTION_COUNT);
+        }
+
+        #[test]
+        fn test_with_gas_offset() {
+            let mut call = ContractCall::new_with_random_id();
+            call.call_parameters = call.call_parameters.with_gas_forwarded(0);
+            let instructions_len = compute_calls_instructions_len(&[call]).unwrap();
+            assert_eq!(
+                instructions_len,
+                Instruction::SIZE * (BASE_INSTRUCTION_COUNT + 2)
```

Comment:
We should use this opportunity to have these tests act as documentation for the inner workings of the SDK/client. Why `+ 2` here and `+ 4` down there? We should have this info as comments right here.